Modules/_datetimemodule.c

```diff
@@ -2284,21 +2284,33 @@ delta_bool(PyDateTime_Delta *self)
 static PyObject *
 delta_repr(PyDateTime_Delta *self)
 {
-    if (GET_TD_MICROSECONDS(self) != 0)
-        return PyUnicode_FromFormat("%s(%d, %d, %d)",
-                                    Py_TYPE(self)->tp_name,
-                                    GET_TD_DAYS(self),
-                                    GET_TD_SECONDS(self),
-                                    GET_TD_MICROSECONDS(self));
-    if (GET_TD_SECONDS(self) != 0)
-        return PyUnicode_FromFormat("%s(%d, %d)",
-                                    Py_TYPE(self)->tp_name,
-                                    GET_TD_DAYS(self),
-                                    GET_TD_SECONDS(self));
-
-    return PyUnicode_FromFormat("%s(%d)",
+    char days[64] = "", seconds[64] = "", microseconds[64] = "";
+    if (GET_TD_DAYS(self) != 0) {
+        sprintf(days, "days=%d", GET_TD_DAYS(self));
+    }
+
+    if (GET_TD_SECONDS(self) != 0 ||
+            (GET_TD_DAYS(self) == 0 && GET_TD_MICROSECONDS(self) == 0)) {
+        if (strlen(days) == 0) {
+            sprintf(seconds, "seconds=%d", GET_TD_SECONDS(self));
+        } else {
+            sprintf(seconds, ", seconds=%d", GET_TD_SECONDS(self));
+        }
+    }
+
+    if (GET_TD_MICROSECONDS(self) != 0) {
+        if (strlen(days) == 0 && strlen(seconds) == 0) {
+            sprintf(microseconds, "microseconds=%d", GET_TD_MICROSECONDS(self));
+        } else {
+            sprintf(microseconds, ", microseconds=%d", GET_TD_MICROSECONDS(self));
+        }
+    }
+
+    return PyUnicode_FromFormat("%s(%s%s%s)",
                                 Py_TYPE(self)->tp_name,
```

Comment:
`tp_name` [contains the fully qualified name](https://github.com/musically-ut/cpython/blob/94d5a4e4d33a1d14a2bb1be8fbff5e1e4cd2b7a6/Modules/_datetimemodule.c#L2444).